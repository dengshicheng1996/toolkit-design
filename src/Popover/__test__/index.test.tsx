import * as React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import {
  ScanCodeOutline,
  SystemQRcodeOutline,
  InformationCircleOutline,
  MoreOutline,
} from '@ant-design/mobile-icons'
import { Popover } from '..'

const Item = Popover.Item

describe('Popover', () => {
  it('snapshot', () => {
    const component = shallow(
      <Popover
        mask
        visible={true}
        overlay={[
          <Item key="4" value="scan" icon={ScanCodeOutline} data-seed="logId">
            Scan
          </Item>,
          <Item key="5" value="special" icon={SystemQRcodeOutline}>
            My Qrcode
          </Item>,
          <Item key="6" value="button ct" icon={InformationCircleOutline}>
            <span style={{ marginRight: 5 }}>Help</span>
          </Item>,
        ]}
        align={{
          overflow: { adjustY: 0, adjustX: 0 },
          offset: [-10, 0],
        }}
        onVisibleChange={() => null}
        onSelect={() => null}
        data-y="111"
      >
        <div>
          点击 <MoreOutline />
        </div>
      </Popover>,
    )

    expect(toJSON(component)).toMatchSnapshot()
  })
})
