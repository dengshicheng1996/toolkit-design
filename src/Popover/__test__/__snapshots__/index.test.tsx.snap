// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover snapshot 1`] = `
<Tooltip
  align={
    Object {
      "offset": Array [
        -10,
        0,
      ],
      "overflow": Object {
        "adjustX": 0,
        "adjustY": 0,
      },
    }
  }
  arrowContent={null}
  data-y="111"
  destroyTooltipOnHide={false}
  mask={true}
  onVisibleChange={[Function]}
  overlay={
    <div
      className="amd-popover-inner-wrapper"
    >
      <PopoverItem
        data-seed="logId"
        disabled={false}
        firstItem={true}
        icon={[Function]}
        onPress={[Function]}
        value="scan"
      >
        Scan
      </PopoverItem>
      <PopoverItem
        disabled={false}
        firstItem={false}
        icon={[Function]}
        onPress={[Function]}
        value="special"
      >
        My Qrcode
      </PopoverItem>
      <PopoverItem
        disabled={false}
        firstItem={false}
        icon={[Function]}
        onPress={[Function]}
        value="button ct"
      >
        <span
          style={
            Object {
              "marginRight": 5,
            }
          }
        >
          Help
        </span>
      </PopoverItem>
    </div>
  }
  placement="bottomRight"
  prefixCls="amd-popover"
  visible={true}
>
  <div>
    点击 
    <MoreOutline
      size="md"
    />
  </div>
</Tooltip>
`;
